packages/dev/core/src/XR/features/WebXRRawCameraAccess.ts

```diff
@@ -0,0 +1,224 @@
+import { WebXRFeatureName, WebXRFeaturesManager } from "../webXRFeaturesManager";
+import type { WebXRSessionManager } from "../webXRSessionManager";
+import { WebXRAbstractFeature } from "./WebXRAbstractFeature";
+import { Observable } from "../../Misc/observable";
+import { Constants } from "../../Engines/constants";
+import { WebGLHardwareTexture } from "../../Engines/WebGL/webGLHardwareTexture";
+import { InternalTexture, InternalTextureSource } from "../../Materials/Textures/internalTexture";
+import { BaseTexture } from "core/Materials/Textures/baseTexture";
+
+/**
+ * Options for raw camera access
+ */
+export interface IWebXRRawCameraAccessOptions {
+    /**
+     * Keep the created textures and metadata when detaching the feature.
+     */
+    doNotDisposeOnDetach?: boolean;
+}
+
+/**
+ * WebXR Feature for WebXR raw camera access
+ * @since
+ * @see https://immersive-web.github.io/raw-camera-access/
+ */
+export class WebXRRawCameraAccess extends WebXRAbstractFeature {
+    private _cachedInternalTextures: InternalTexture[] = [];
+    /**
+     * This function will return an array of camera views
+     * Note that mostly the array will contain a single view
+     * If you want to know the order of the views, use the `viewIndex` property
+     * @returns the array of base textures which can be used throughout the framework

```

Comment:
Is this comment the right one, as `texturesData` is not a function(?)

Reply:
it was before a restructure. thanks!! will fix it.